Bloom.md

```diff
@@ -0,0 +1,248 @@
+
+The data above is written into the RDB during save operations and can be restored from RDB load. However, the data that gets written to the RDB is specific to the data type's structure and struct members. Additionally, the data within the underlying bloom filter (from the external crate) is specific to the implementation of the bloom filter as the hash key (seed), raw bit array data, etc. can all vary.
+
+Because of this, it is not possible to be RDB compatible with ReBloom. It might be possible if the RDB Save/Load of the ReBloom module is reverse engineered AND also the bloom filter implementation is changed to follow ReBloom Module's raw Bloom Filter structure syntax. Additionally, if there is any RDB change in ReBloom, this reverse engineering + structure updates will be needed once again.
+
+For these reasons, ValkeyBloom is not RDB Compatible with ReBloom.
+
+### AOF Rewrite handling:
+
+Module data types (including bloom) can implement a callback function that will be triggered for Bloom objects to rewrites its data as command/s. From the AOF callback, we can handle AOF rewrite by saving a RESTORE command with the key, TTL, and serialized value of the corresponding bloom object.
+
+There is an alternate to the approach above. It involves having the ValkeyBloom Module support BF.LOADCHUNK and BF.SCANDUMP commands and the AOF Rewrite callback can write BF.LOADCHUNK commands to restore a bloom object in interations. This approach was not chosen because we can already save and restore bloom object data from RDB data using the RESTORE command without having to support additional Module commands.
+
+### Migrating workloads from ReBloom:
+
+Customers that currently use ReBloom can move to ValkeyBloom using two approaches:
+
+1. Create the bloom filter objects to have the same properties using BF.RESERVE / BF.INSERT on Valkey (with ValkeyBloom loaded). Re-populate the bloom filter objects by inserting items by moving the existing bloom workload to the Valkey server (with ValkeyBloom loaded). The workload can be moved without any errors since we are API Compatible.
+
+Pros
+* This is the simplest option in terms of effort - assuming the user is fine with recreating bloom objects and adding items on the Valkey server (with ValkeyBloom).
+
+Cons
+* The user will need to re-create the bloom objects (using BF.RESERVE/BF.INSERT) and populate these objects **AFTER** switching to Valkey (with ValkeyBloom).
+
+2. Users can generate an AOF file from a server (that has the ReBloom module loaded) and with an on-going bloom workload that creates bloom filters & inserts items into them. Next, this can be re-played on a Valkey Server (with ValkeyBloom loaded). Then, the user can move their existing bloom workload to the Valkey server (with ValkeyBloom loaded).
```

Comment:
AOF will eventually need to be truncated, I don't think this is really a tenable approach for migration. Bloom filters are intended to be lossy.

Maybe we should instead consider a way to load a bloom module with its hashes, but use our own hash algorithm when we are generating our own? We might need new types of module APis for that though.

Reply:
>AOF will eventually need to be truncated, I don't think this is really a tenable approach for migration.

I agree, this will not work when there are truncations in AOF or rewrites.

>Bloom filters are intended to be lossy.

This is true. Customers also use bloom filters with TTLs and these customers should be ok with recreating bloom objects after migrating to Valkey